onnx_chainer/graph.py

```diff
@@ -0,0 +1,179 @@
+
+            for input_ in creator.inputs:
+                if input_ is creator:
+                    continue
+                if (input_, creator) in seen_edges:
+                    continue
+                add_cand(input_)
+                seen_edges.add((input_, creator))
+
+        return reversed(function_nodes.values())
+
+    def create_node(
+            self, func_name, func, input_names, output_names, parameters):
+        onnx_helper.set_func_name(func_name)
+        converter = self.converters.get(func_name, None)
+        if converter is None:
+            raise ValueError('{} is not supported'.format(func_name))
+        params = FunctionConverterParams(
+            func, self.specified_opset_version, input_names, output_names,
+            self.context, parameters)
+        nodes = converter(params)
+        nodes = list(reversed(nodes))
+        assert len(nodes[0].output) == len(output_names)
+        nodes[0].output[:] = output_names
+        return nodes
+
+    def convert_to_onnx_node(self, function):
+        if isinstance(function, chainer.function.FunctionAdapter):
+            function = function.function
+        func_name = getattr(
+            function, 'custom_function_node_name', function.__class__.__name__)
+        temp_node_name = '{}:{}'.format(
+            self.func_name_counts[func_name], func_name)
+        self.func_name_counts[func_name] += 1
+
+        input_names = []
+        for input_var in function.inputs:
+            # 'input_var' is a VariableNode,
+            # so check if it has a Variable/Parameter
+            var = input_var.get_variable_or_none()
+            if var is None:  # No reference to Variable/Parameter
+                # Use VariableNode as is
+                input_name = self.context.get_name(input_var)
+            else:  # It is a parameter inside a Link or network input
+                input_name = self.context.get_name(var)
+                self.inputs[input_name] = var
+            input_names.append(input_name)
+
+        # This is to get corresponding VariableNode id from the output
+        # Variable of the network
+        output_names = []
+        for output_ref in function.outputs:
+            if output_ref() is None:
+                output_name = self.context.get_name(output_ref)
+            else:
+                var = output_ref().get_variable_or_none()
+                if var is not None:  # If the output is kept
+                    output_name = self.context.get_name(var)
+                    if output_name in self.inputs:
+                        del self.inputs[output_name]
+                else:
+                    output_name = self.context.get_name(output_ref())
+            output_names.append(output_name)
+
+        nodes = self.create_node(
+            func_name, function, input_names, output_names,
+            self.additional_parameters)
+        self.converted_nodes[temp_node_name] = nodes
+
+    def after_rename(self):
```

Comment:
I think `rename_outputs` would be a better name. I guess by "after" you mean this is a postprocessing path, but `after_rename` sounds a bit strange. I think "rename" would imply we are in postprocessing.